src/lib/lock-utils.ts

```diff
@@ -0,0 +1,59 @@
+import { UniqueConstraintError } from 'sequelize';
+import { wait } from 'ts-retry-promise';
+import { Lock } from '../models/Lock';
+
+const DEFAULT_TIMEOUT = 10 * 1000;
+const DEFAULT_POOL_INTERVAL = 500;
+
+type Options = {
+  timeout?: number;
+  targetPollInterval?: number;
+};
+const DefaultOptions = {
+  timeout: DEFAULT_TIMEOUT,
+  targetPollInterval: DEFAULT_POOL_INTERVAL,
+};
+
+export class LockTimeoutError extends Error {
+  constructor(message?: string) {
+    const name = 'LockTimeoutError';
+    super(name + ' ' + (message || '').trim());
+    this.name = name;
+  }
+}
+
+// inspired by https://github.com/marcosschroh/yoyo-database-migrations/blob/master/yoyo/backends.py#L305
+export async function acquireLock(key: string, options?: Options) {
+  const { timeout, targetPollInterval } = { ...DefaultOptions, ...options };
+
+  const pollInterval = Math.min(targetPollInterval, timeout);
+  const startedDate = Date.now();
+
+  while (true) {
+    try {
+      await Lock.create({
+        key,
+      });
```

Comment:
In context of transaction, when first process acquires lock by inserting, next process hangs here.
Else, when first process acquires lock by inserting, next process will fail and we'll retry on interval till timeout is overpassed.